.gitlab/scripts/git_try_to_push

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+# at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+# LICENSE and NOTICE for details. LLNL-CODE-806117.
+#
+# This file is part of the MFEM library. For more information and source code
+# availability visit https://mfem.org.
+#
+# MFEM is free software; you can redistribute it and/or modify it under the
+# terms of the BSD-3 license. We welcome feedback and contributions, see file
+# CONTRIBUTING.md for details.
+
+# Try to push to the remote 5 times. If the push fails, and the local and remote
+# have diverged, then pull from the remote to merge changes, and try pushing
+# again. If some other failure happens
+for i in {1..5}; do
+   git push origin master
+   if [[ $? == 0 ]]; then
+      exit 0 # success
+   else
+   # Fetch any updates from the remote
```

Comment:
I added a 20 second delay.

I would hope that a delay isn't really necessary, i.e. that if `git push` fails because the remote has diverged, then the subsequent `git pull` would fix things without having a delay, but I don't know enough about how git works internally to know for sure.

Reply:
I was thinking that it's good to wait a bit before trying to connect to the server again, so to me it makes sense to wait before `git remote update`. However, as you said, the wait is probably not crucial but more like a precaution.